SPECS/python3/python3.spec

```diff
@@ -163,6 +163,8 @@ export OPT="%{extension_cflags} %{openssl_flags}"
 # and install fails. We will install these two bundled wheels manually.
 # https://github.com/pypa/pip/issues/3063
 # https://bugs.python.org/issue31916
+# Bootstrap pip3 which casues ptest build failure
+./python Lib/ensurepip
```

Comment:
Just for my own knowledge: how does this work? The line below suggests that `Lib/ensurepip` is a directory, so I'm curious what's happening here.

Reply:
Also, could you extend the summary a little bit? I'm not sure what ptests is this change fixing, since this spec has got the `%check` section currently commented out.

Reply:
Sure, let me try to explain what I understood from the weird error that was coming up from me trying to build it locally.

Yep, the check is commented here, but the ptest pipeline, builds python3 and in course of doing it, breaks :( right where it wants to `install pip using pip`, which it provides as well :)

Our python3 package also provides pip3, and in order to install pip3 from the available sources, we need pip3.
the line `python Lib/ensurepip`  runs the module `ensurepip` which has a `__init__.py` file, which makes it a python module, which can be imported/run.
The line `./python Lib/ensurepip` runs the module, which installs pip3 which is then used in the appropriate way to install the bundled `pip3 & setuptools`
ALthough, I would like @thcrain-msft to double check what i've said :) and `ensuresanity`

Reply:
I believe everything Falak said is correct. The manual installation of pip into the RPM buildroot requires pip to already be present in the build chroot. This is satisfied by the raw toolchain version of Python 3 during toolchain builds.

This is a good workaround. The added line is essentially a no-op in the toolchain, so things are fine there. The only cleaner alternate solution I see is adding a build requirement on `python3-pip`, but I honestly don't know if that would cause `python3` to rebuild in non-ptest builds.

Reply:
Thanks guys, sounds good!